sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs

```diff
@@ -257,18 +295,26 @@ internal static BlockBlobStageBlockOptions ToBlobStageBlockOptions(this BlockBlo
         internal static BlobSyncUploadFromUriOptions ToSyncUploadFromUriOptions(
             this BlockBlobStorageResourceOptions options,
             bool overwrite,
-            HttpAuthorization sourceAuthorization)
+            HttpAuthorization sourceAuthorization,
+            StorageResourceItemProperties resourceProperties = default)
         {
             // There's a lot of conditions that cannot be applied to a Copy Blob (async) Request.
             // We need to omit them, but still apply them to other requests that do accept them.
             // See https://learn.microsoft.com/en-us/rest/api/storageservices/copy-blob-from-url#request-headers
             // to see what headers are accepted.
+            AccessTier? accessTier = default;
+            if (resourceProperties != default)
+            {
+                resourceProperties.RawProperties.TryGetValue(DataMovementConstants.ResourceProperties.AccessTier, out object tier);
+                accessTier = (AccessTier?)tier;
+            }
```

Comment:
Not sure if you changed anything here but I meant to revert the changes to `ToSyncUploadFromUriOptions` for now and do them later since there is more work that is needed?

Reply:
Removed this, will add this back in separate PR when it's actually being used.